Use Wiremock to simulate the request/response of the MS Graph API

amitjangra2599 opened the pull request:
# Description

Use Wiremock to simulate the request/response of the MS Graph API.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Database change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Unit Tests
- [x] Integration Tests
- [ ] Manual Testing (please describe)

## Checklist:

- [ ] I have created ADR (Architecture Decision Record) for this change, in case of architecture or tech stack changes.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes in other branches have been merged
- [x] I have updated the relevant application properties files (if needed)
- [ ] I have updated any necessary database migrations (if needed)

## Screenshots (for UX changes):

## Additional Notes:

Please add any additional notes, dependencies, or context that reviewers should know about.

amitjangra2599 committed:
Use Wiremock to simulate the request/response of the MS Graph API

amitjangra2599 commented:
Hi @axymthr, I’ve implemented the WireMock simulation for pulling employees using MS Graph.
Whenever you get a chance, please review it and let me know if you have any feedback or suggestions.

axymthr commented:
I'm not clear on the intent of this PR. Are you going to raise several PRs for this feature? All you have here is a test scaffolding and an API call where the response isn't being used.

amitjangra2599 committed:
Moved access token logic into reusable AccessTokenService
- Integrated sync with Microsoft Graph API to fetch employee data
- Added @Scheduled task to run sync every 24 hours

amitjangra2599 commented:
Hi @akshay-nanda-sneo I have added more work in this PR. Please review and let me know any  if you have any feedback or suggestions.
